Ability Controller to help with RDX

EAOZONE opened the pull request.

EAOZONE committed:
makes AbilityHandController moving a lot of RDXAbilityHand away from it to there

TomaszTB reviewed (commented):
I like this, but I think the API could be better. What about having methods like `setGoalPosition`, `setGoalVelocity`, and `setControlMode`? If `controlMode == POSITION` it'll only use the goal position values and send the position command. If `controlMode == VELOCITY` it'll only use the goal velocity values and send the command. If `controlMode == VELOCITY_TO_POSITION` it'll use both position and velocity values to do its thing. 

Also, I'd put everything that has to happen repetitively in the `update` method. E.g. if it's in the velocity to position control mode, the `processVelToPos` method can be called in the update method.

Lastly, I'm not sure about this class having a communication class of its own. I imagined that one instance of the communication class would be shared by two hands. Not sure how well that design works though

EAOZONE committed:
changes some of Tomasz's suggestsions

EAOZONE commented:
Made all of the getters and setters, made both POSTION and VELOCITY and controlMode, moved everything that is being called every tick to update and removed the instance's of the communication class in both AbilityHandController and RDXAbilityHand

EAOZONE committed:
Implements way to move final actuator even if fingers don't make it to the final position

EAOZONE committed:
Fixes hand connection at beginning for rotator to be in negative position

EAOZONE committed:
Fixes error where hand would keep moving with velocity even when controlling position

TomaszTB commented:
Took these changes, improved on them, and merged in #12
